pkg/report/report.go

```diff
@@ -465,105 +445,11 @@ func Generate(ctx context.Context, path string, mrs *yarax.ScanResults, c malcon
 		ruleURL := generateRuleURL(m.Namespace(), m.Identifier())
 
-		var matchedStrings []string
-		{
-			totalMatches := 0
-			for _, p := range m.Patterns() {
-				totalMatches += len(p.Matches())
-			}
-
-			matches := make([]yarax.Match, 0, totalMatches)
-			for _, p := range m.Patterns() {
-				matches = append(matches, p.Matches()...)
-			}
-
-			processor := newMatchProcessor(fc, matches, m.Patterns())
-			matchedStrings = processor.process()
-		}
-
-		b := &malcontent.Behavior{
-			ID:           key,
-			MatchStrings: matchStrings(m.Identifier(), matchedStrings),
-			RiskLevel:    RiskLevels[risk],
-			RiskScore:    risk,
-			RuleName:     m.Identifier(),
-			RuleURL:      ruleURL,
-		}
-
-		k := ""
-		v := ""
+		matchedStrings := processMatchedStrings(fc, m)
 
-		for _, meta := range m.Metadata() {
-			k = meta.Identifier()
-			v = fmt.Sprintf("%s", meta.Value())
-			// Empty data is unusual, so just ignore it.
-			if k == "" || v == "" {
-				continue
-			}
-
-			// If we find a match in the map for the metadata key, that's the rule to override
-			// Store this rule (the override) in the fr.Overrides behavior slice
-			// If an override rule is not overriding a valid rule, log an error
-			_, exists := mrsMap[k]
-			switch {
-			case exists && override:
-				var overrideSev int
-				if sev, ok := Levels[v]; ok {
-					overrideSev = sev
-				}
-				b.RiskLevel = RiskLevels[overrideSev]
-				b.RiskScore = overrideSev
-				b.Override = append(b.Override, k)
-				fr.Overrides = append(fr.Overrides, b)
-			case !exists && override:
-				// TODO: return error if override references an unknown rule name
-				continue
-			}
+		b := buildBehavior(m, matchedStrings, key, ruleURL, risk)
 
-			switch k {
-			case "author":
-				b.RuleAuthor = v
-				m := authorWithURLRe.FindStringSubmatch(v)
-				if len(m) > 0 && isValidURL(m[2]) {
-					b.RuleAuthor = m[1]
-					b.RuleAuthorURL = m[2]
-				}
-				// If author is in @username format, strip @ to avoid constantly pinging them on GitHub
-				if strings.HasPrefix(b.RuleAuthor, "@") {
-					b.RuleAuthor = strings.Replace(b.RuleAuthor, "@", "", 1)
-				}
-			case "author_url":
-				b.RuleAuthorURL = v
-			case fmt.Sprintf("__%s__", NAME):
-				if v == "true" {
-					fr.IsMalcontent = true
-				}
-			case "license":
-				b.RuleLicense = v
-			case "license_url":
-				b.RuleLicenseURL = v
-			case "description", "threat_name", "name":
-				desc := mungeDescription(v)
-				if len(desc) > len(b.Description) {
-					b.Description = desc
-				}
-			case "ref", "reference":
-				u := fixURL(v)
-				if isValidURL(u) {
-					b.ReferenceURL = u
-				}
-			case "source_url":
-				// YARAforge forgets to encode spaces
-				b.RuleURL = fixURL(v)
-			case "pledge":
-				pledges = append(pledges, v)
-			case "syscall":
-				sy := strings.Split(v, ",")
-				syscalls = append(syscalls, sy...)
-			case "cap":
-				caps = append(caps, v)
-			}
-		}
+		handleMetadata(m, b, fr, override, mrsMap, pledges, caps, syscalls)
```

Comment:
I noticed we're missing pledges and syscalls (and most likely caps even though we don't currently have any samples with this) in the recent test diffs.

We can probably move `pledges`, `caps`, and `syscalls` to `handleMetadata`'s scope.

To verify your changes locally you can run `make refresh-sample-testdata` to build the latest changes and refresh our sample data. If you see any diffs, you can quickly iterate and try again rather than waiting for CI.